src/routing_table/prefix.rs

```diff
@@ -116,6 +116,38 @@ impl<T: Clone + Copy + Default + Binary + Xorable> Prefix<T> {
     pub fn upper_bound(&self) -> T {
         self.name.set_remaining(self.bit_count, true)
     }
+
+    /// Returns whether the namespace defined by `self` is covered by prefixes in the `prefixes`
+    /// set
+    pub fn is_covered_by<'a, U>(&self, prefixes: U) -> bool
+        where T: 'a,
+              U: IntoIterator<Item = &'a Prefix<T>> + Clone
+    {
+        let max_prefix_len = prefixes.clone().into_iter().map(|x| x.bit_count()).max().unwrap_or(0);
```

Comment:
Because `into_iter()` consumes `self`, so passing `prefixes` to `is_covered_by_impl` in the next line would result in a `use of a moved value`.